components/Header/index.jsx

```diff
@@ -1,66 +1,170 @@
+import Image from "next/image";
+import { AiOutlineClose } from "react-icons/ai";
+import { useState } from "react";
+
+const socialMedia = [
+  { src: Facebook, alt: "Facebook", link: "/" },
+  { src: Instagram, alt: "Instagram", link: "/" },
+  { src: Twitter, alt: "Twitter", link: "/" },
+  { src: GitHub, alt: "GitHub", link: "/" },
+];
+
+const navItems = [
+  { name: "Home", link: "/" },
+  { name: "Eligibility", to: "eligibility" },
+  { name: "Timeline", to: "timeline" },
+  { name: "Gallery", to: "gallery" },
+  { name: "Rules", link: "/rules" },
+];
+
 //
 // TODO: Create a header component that displays the header of the website.
 export default function Header() {
+  const [menuOpen, setMenuOpen] = useState(false);
+
+  const handleNav = () => {
+    setMenuOpen(!menuOpen);
+  };
+
   return (
     // * Example usage of the header component.
     // ! Change as needed.
     <main className="sticky top-0 z-50 bg-white shadow-md">
-      <ul className="flex justify-around py-4 mb-5">
-        <li>
-          <Link href="/" className="text-lg font-semibold">
-            Home
-          </Link>
-        </li>
-        <li>
-          <Scroll
-            className="text-lg font-semibold"
-            to="eligibility"
-            spy={true}
-            smooth={true}
-            offset={10}
-            duration={500}
-            delay={100}
-          >
-            Eligibility
-          </Scroll>
-        </li>
-        <li>
-          <Scroll
-            className="text-lg font-semibold"
-            to="timeline"
-            spy={true}
-            smooth={true}
-            offset={10}
-            duration={500}
-            delay={100}
-          >
-            Timeline
-          </Scroll>
-        </li>
-        {/* Teams - no plans yet */}
-        <li>
-          <Scroll
-            className="text-lg font-semibold"
-            to="gallery"
-            spy={true}
-            smooth={true}
-            offset={10}
-            duration={500}
-            delay={100}
+      <div className="flex flex-col items-center flex-wrap">
+        <Link href="/">
+          <Image
+            src={logomain}
+            alt="logo-main"
+            width="309"
+            height="358"
+            className="cursor-pointer mt-4"
+            priority
+          />
+        </Link>
+        <div className="flex justify-between items-center h-full w-full px-4 md:px-[120px] lg:px-[202px] my-5">
+          {/* nav bar links */}
+          <div className="hidden lg:flex">
+            <ul className="flex justify-around gap-12 items-center">
+              {navItems.map((item, index) => (
+                <li key={index}>
+                  {item.link ? (
+                    <Link href={item.link} className="text-lg font-semibold">
+                      {item.name}
+                    </Link>
+                  ) : (
+                    <Scroll
+                      className="text-lg font-semibold"
```

Comment:
When hovering the text element, the cursor should be a pointer. Use tailwindcss inline styles to add that.